src/include/storage/index/stl_comparator_wrapper.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+namespace bustub {
+
+template <typename KeyType, typename KeyComparator>
+class StlComparatorWrapper {
+ public:
+  explicit StlComparatorWrapper(const KeyComparator &cmp) : cmp_{cmp} {}
+
+  inline auto operator()(const KeyType &lhs, const KeyType &rhs) const -> bool { return cmp_(lhs, rhs) < 0; }
```

Comment:
Oh! I just realized why my wrapper on Friday was seeing weird bugs...

I thought `operator()` would return `int` like a Java comparator, not a `bool` based on `std::less`. So I just used `cmp_(lhs, rhs)` directly. Oops.

Reply:
Yeah, I also feel C++'s compile error is unhelpful and cannot point to the exact place that I got wrong with 😇